applications/OptimizationApplication/python_scripts/responses/response_function.py

```diff
@@ -1,27 +1,115 @@
 from abc import ABC, abstractmethod
+from typing import Union
 import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
 from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
 
 class ResponseFunction(ABC):
-    @classmethod
+    """Base response function.
+
+    This reponse function is the base response function. This is assumed to have following responsibilities.
+        1. CalculateValue for a fresh design. (@see CalculateValue)
+        2. CalculateSensitivity for a fresh design (@see CalculateSensitivity)
+
+    This response should only work on one model part. Hence, if multiple model parts required then,
+    a single model part should be created using Kratos.ModelPartOperationUtilities.
+    """
+    def __init__(self, response_name: str) -> None:
+        self.__name = response_name
+
+    def GetName(self) -> str:
+        return self.__name
+
     @abstractmethod
-    def GetSensitivityFieldVariables(cls) -> 'list[SupportedSensitivityFieldVariableTypes]':
+    def Initialize(self) -> None:
+        """Initializes the response.
+
+        This method initializes the response. This is only called once in the whole optimization process.
+
+        """
         pass
 
-    def Initialize(self) -> None:
+    @abstractmethod
+    def Check(self) -> None:
+        """Checks the response.
+
+        This method checks the response. This is only called once in the whole optimization process.
+
+        """
         pass
 
+    @abstractmethod
     def Finalize(self) -> None:
+        """Finalizes the response.
+
+        This method finalizes the response. This is only called once in the whole optimization process.
+
+        """
         pass
 
     @abstractmethod
-    def Check(self) -> None:
+    def GetDependentPhysicalKratosVariables(self) -> 'list[SupportedSensitivityFieldVariableTypes]':
```

Comment:
```suggestion
    def GetImplementedPhysicalKratosVariables(self) -> 'list[SupportedSensitivityFieldVariableTypes]':
```

Reply:
After one more round of discussions we get to a decision:
1. We return here only the list of the des_var that are implemneted for Sens. Analysis
2. We add the Messages into log file, where we assume sens are zero
3. All raw sens can be outputted into the vtk vtu etc. for visual check
4. We go with suggested name above: GetImplementedPhysicalKratosVariables